spring-xd-messagebus-rabbit/src/main/java/org/springframework/xd/dirt/integration/rabbit/RabbitMessageBus.java

```diff
@@ -673,6 +677,34 @@ public void destroy() {
 		stopBindings();
 	}
 
+	@Override
+	public void storeForManualAck(MessageHeaders headers) {
+		if (headers.containsKey(AmqpHeaders.CHANNEL)) {
+			Channel channel = (com.rabbitmq.client.Channel) headers.get(AmqpHeaders.CHANNEL);
+			Long deliveryTag = (Long) headers.get(AmqpHeaders.DELIVERY_TAG);
+			if (channelsForManualAck.containsKey(channel) && (channelsForManualAck.get(channel) < deliveryTag)) {
+				channelsForManualAck.put(channel, deliveryTag);
+			}
+			else {
+				channelsForManualAck.put(channel, deliveryTag);
+			}
+		}
+	}
+
+	@Override
+	public void doManualAck() {
+		for (Map.Entry<Object, Long> entry : channelsForManualAck.entrySet()) {
```

Comment:
That's a great catch. We do have access to `streamBlockId` for each block being pushed. I can add blockId as a key to each batch of messages that get acked.